src/util.ts

```diff
@@ -789,8 +791,8 @@ export class Util {
      * @param array the array to flatMap over
      * @param callback a function that is called for every array item
      */
-    public flatMap<T, R>(array: T[], callback: (arg: T) => R) {
-        return Array.prototype.concat.apply([], array.map(callback)) as never as R;
+    public flatMap<T, R>(array: T[], callback: (arg: T) => R[]): R[] {
+        return Array.prototype.concat.apply([], array.map(callback));
```

Comment:
This is the same type, spelled differently. In the original `R` was assumed to be an array of some other type, whereas in the new version the array-ness of the return type is explicit. This improves type inference at a usage site.